File: mobile-19t/apps/mobile/lib/features/hr/hr_role_utils.dart
const hrManagerRoleId = '4a5ce6f3-25e9-462a-b161-439a6a4e3e99';
const employeeManagementRoleIds = <String>{
  '4a5ce6f3-25e9-462a-b161-439a6a4e3e99',
};

bool canApproveLeavesForRoles(List<String> roles) {
  return roles.contains('admin') ||
      roles.contains('manager') ||
      roles.contains(hrManagerRoleId);
}

bool canManageEmployeesForRoles(List<String> roles) {
  return roles.contains('admin') ||
      roles.contains('manager') ||
      roles.any(employeeManagementRoleIds.contains);
}

bool canViewEmployeeAttendanceForRoles(List<String> roles) {
  return canManageEmployeesForRoles(roles);
}

bool canEditOwnHrProfileForRoles(List<String> roles) {
  return roles.isNotEmpty;
}

File: mobile-19t/apps/api/src/modules/hr/hr-role.utils.ts
export const HR_MANAGER_ROLE_ID = '4a5ce6f3-25e9-462a-b161-439a6a4e3e99';

const HR_LEAVE_APPROVER_ROLES = new Set(['admin', 'manager', HR_MANAGER_ROLE_ID]);

export function canApproveLeaveRoles(roles?: string[] | null): boolean {
  return (roles ?? []).some((role) => HR_LEAVE_APPROVER_ROLES.has(role));
}
